website/static/js/fileBrowser.js

```diff
@@ -0,0 +1,1175 @@
+            self.activityLogs([]); // Empty old logs first;
+            self.showMoreActivityLogs(null);
+            promise.then(function _getLogsThen (result){
+                result.data.map(function(log){
+                    log.attributes.formattableDate = new $osf.FormattableDate(log.attributes.date);
+                    if(addToExistingList){
+                        self.activityLogs().push(log);
+                    }
+                });
+                if(!addToExistingList){
+                    self.activityLogs(result.data);  // Set activity log data
+                }
+                self.showMoreActivityLogs(result.links.next); // Set view for show more button
+            });
+            return promise;
+        };
+        // separate concerns, wrap getlogs here to get logs for the selected item
+        self.getCurrentLogs = function _getCurrentLogs ( ){
+            if(self.selected().length === 1){
+                var id = self.selected()[0].data.id;
+                var promise = self.getLogs(id);
+            }
+        };
+
+        /* filesData is the link that loads tree data. This function refreshes that information. */
+        self.updateFilesData = function _updateFilesData (linkObject) {
+            if (linkObject.link !== self.currentLink) {
+                self.updateBreadcrumbs(linkObject);
+                self.updateList(linkObject);
+                self.currentLink = linkObject.link;
+            }
+            self.showSidebar(false);
+        };
+
+        // INFORMATION PANEL
+        /* Defines the current selected item so appropriate information can be shown */
+        self.selected = m.prop([]);
+        self.updateSelected = function _updateSelected (selectedList){
+            self.selected(selectedList);
+            self.getCurrentLogs();
+        };
+
+        // USER FILTER
+        self.activeFilter = m.prop({});
+        self.updateFilter = function _updateFilter (filter) {
+            self.activeFilter(filter);
+            self.updateFilesData(filter);
+        };
+
+        self.removeProjectFromCollections = function _removeProjectFromCollection () {
+            // Removes selected items from collect
+            var collection = self.activeFilter().data.node;
+            self.selected().map(function _removeProjectFromCollectionsMap (item){
+                m.request({
+                    method : 'DELETE',
+                    url : collection.links.self + 'node_links/' + item.data.id + '/',
+                    config : xhrconfig
+                }).then(function _removeProjectFromCollectionsSuccess(result){
+                    console.log(result);
+                }, function _removeProjectFromCollectionsFail(result){
+                    console.log(result);
+                });
+            });
+        };
+        // GETTING THE NODES
+        self.updateList = function _updateList (linkObject, success, error){
```

Comment:
In general, promises are preferred to `success` and `error` parameters. That said, `success` and `error` don't appear to be used anywhere in the code. If you can confirm that, please remove them.